api/app/src/command/medical/document/check-doc-queries.ts

```diff
@@ -0,0 +1,175 @@
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { QueryTypes } from "sequelize";
+import { Progress } from "../../../domain/medical/document-reference";
+import { Patient, PatientModel } from "../../../models/medical/patient";
+import { capture } from "../../../shared/notifications";
+import { Util } from "../../../shared/util";
+import {
+  MAPIWebhookStatus,
+  MAPIWebhookType,
+  processPatientDocumentRequest,
+} from "../../webhook/medical";
+import { appendDocQueryProgress } from "../patient/append-doc-query-progress";
+
+dayjs.extend(duration);
+
+const MAX_CONVERSION_TIME = dayjs.duration({ minutes: 30 });
+const MAX_DOWNLOAD_TIME = dayjs.duration({ minutes: 30 });
+
+/**
+ * Ops-driven function to check the status of all document queries in progress.
+ * NOT TO BE USED BY CUSTOMERS DIRECTLY.
+ *
+ * For each patient with a document query in progress that match the criteria,
+ * it will update the status to `completed` and trigger the webhook.
+ *
+ * @param patientIds - Optional list of patient IDs to check. If not provided, all patients will be checked.
+ */
+export async function checkDocumentQueries(patientIds: string[]): Promise<void> {
+  const { log } = Util.out(`checkDocumentQueries - patientIds ${patientIds.join(", ")}`);
+  try {
+    const patientsWithDownloadsProcessing = await getDownloadsToUpdate(patientIds);
+    log(
+      `Got ${
+        patientsWithDownloadsProcessing.length
+      } patients with downloads 'processing' to be moved to 'completed': ${patientsWithDownloadsProcessing.join(
+        ", "
+      )}`
+    );
+    for (const patient of patientsWithDownloadsProcessing) {
+      await appendDocQueryProgress({
+        patient,
+        downloadProgress: {
+          status: "completed",
+        },
+      });
+      processPatientDocumentRequest(
+        patient.cxId,
+        patient.id,
+        MAPIWebhookType.documentDownload,
+        MAPIWebhookStatus.completed
+        // Can we send these without the list of documents?
+        // Concerned with either:
+        // - doing one patient at a time and taking so long that rolls over the next execution; OR
+        // - doing all patients at once and hammering the FHIR server
+        // toDTO(docsNewLocation)
+      );
+    }
+
+    const patientsWithConversionsInProgress = await getConversionsToUpdate(patientIds);
+    log(
+      `Got ${
+        patientsWithConversionsInProgress.length
+      } patients with conversions 'processing' to be moved to 'completed': ${patientsWithConversionsInProgress.join(
+        ", "
+      )}`
+    );
+    for (const patient of patientsWithConversionsInProgress) {
+      await appendDocQueryProgress({
+        patient,
+        convertProgress: {
+          status: "completed",
+        },
+      });
+      processPatientDocumentRequest(
+        patient.cxId,
+        patient.id,
+        MAPIWebhookType.documentConversion,
+        MAPIWebhookStatus.completed
+      );
+    }
```

Comment:
I'm thinking we should also send the list of updated patients (both for `download` and `convert`) to Sentry so we keep track of it, right?

Reply:
Sentry notification added